sklearn/manifold/t_sne.py

```diff
@@ -628,7 +632,7 @@ class TSNE(BaseEstimator):
     def __init__(self, n_components=2, perplexity=30.0,
                  early_exaggeration=12.0, learning_rate=200.0, n_iter=1000,
                  n_iter_without_progress=300, min_grad_norm=1e-7,
-                 metric="euclidean", init="random", verbose=0,
+                 metric="euclidean", p=2, init="random", verbose=0,
```

Comment:
You've not changed the default for metric. But if we make this change in tSNE, we would also need to square distances where metric='minkowski' and p=2. See also #11793, #12401

Reply:
Hi @jnothman 

When I check the `metric='minkowski'` and the `p=2` I'll do the calculation through [this](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.euclidean_distances.html) function?

[Here](https://github.com/scikit-learn/scikit-learn/blob/f0ab589f1541b1ca4570177d93fd7979613497e3/sklearn/manifold/t_sne.py#L686) I'll include a new `if` for a new metric? That's it?

Thank you in advance!

Reply:
@gilbertoolimpio  For the time being, you can revert your changes to tsne, and focus on the other two. I guess those are more straightforward to handle.

Reply:
@adrinjalali 
Ok, I'll revert this changes.